packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -321,9 +321,13 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     const { iconProps } = this.props;
 
     if (iconProps && (iconProps.iconName !== undefined || iconProps.imageProps)) {
-      const { className, ...rest } = iconProps;
+      const { className, imageProps, ...rest } = iconProps;
 
-      return <Icon className={css(this._classNames.icon, className)} {...rest} />;
+      if (iconProps.iconName) {
```

Comment:
You might want to add a check for `!iconProps.styles` before using the new icons.

Reply:
I mean use the old `Icon` if the icon has styles